overlord/ifacestate/handlers.go

```diff
@@ -711,6 +711,21 @@ func checkDisconnectConflicts(st *state.State, disconnectingSnap, plugSnap, slot
 	return nil
 }
 
+// inWaitChains returns true if there is a wait chain so that t1
+// is run before needle
+func inWaitChain(t1, needle *state.Task) bool {
+	for _, cand := range t1.HaltTasks() {
```

Comment:
we should check first that t1 and needle are in the same change otherwise we can return false quickly, we probably want to do it in a top level wrapper though, no point in doing it for each recursion